core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/EraSubmissionPreparation.java

```diff
@@ -0,0 +1,496 @@
+
+    @Override
+    protected void populate(Element xml, Study s) {
+      xml.setAttribute("alias", s.getAlias());
+
+      Element studyDescriptor = xml.getOwnerDocument().createElementNS(null, "DESCRIPTOR");
+      xml.appendChild(studyDescriptor);
+
+      Element studyTitle = xml.getOwnerDocument().createElementNS(null, "STUDY_TITLE");
+      studyTitle.setTextContent(s.getAlias());
+      studyDescriptor.appendChild(studyTitle);
+
+      Element studyType = xml.getOwnerDocument().createElementNS(null, "STUDY_TYPE");
+      studyType.setAttribute("existing_study_type", s.getStudyType().getName());
+      studyDescriptor.appendChild(studyType);
+
+      Element centerProjectName = xml.getOwnerDocument().createElementNS(null, "CENTER_PROJECT_NAME");
+      centerProjectName.setTextContent(s.getProject().getAlias());
+      studyDescriptor.appendChild(centerProjectName);
+
+      Element studyAbstract = xml.getOwnerDocument().createElementNS(null, "STUDY_ABSTRACT");
+      studyDescriptor.appendChild(studyAbstract);
+
+      Element studyDescription = xml.getOwnerDocument().createElementNS(null, "STUDY_DESCRIPTION");
+      studyDescription.setTextContent(s.getDescription());
+      studyDescriptor.appendChild(studyDescription);
+    }
+
+    @Override
+    protected String xmlCollectionName() {
+      return "STUDY_SET";
+    }
+
+    @Override
+    protected String xmlName() {
+      return "STUDY";
+    }
+  }
+
+  private abstract class XmlSubmissionFromSet<T> extends ChildSubmissionFile {
+    @Override
+    public void generateDocument(ZipOutputStream output)
+        throws ParserConfigurationException, IOException, TransformerConfigurationException, TransformerException,
+        TransformerFactoryConfigurationError {
+      Document document = DocumentBuilderFactory.newInstance().newDocumentBuilder().newDocument();
+
+      Element set = document.createElementNS(null, xmlCollectionName());
+      document.appendChild(set);
+      items().map(item -> {
+        Element itemElement = document.createElement(xmlName());
+        populate(itemElement, item);
+        return itemElement;
+      }).forEach(set::appendChild);
+
+
+      output.putNextEntry(new ZipEntry(fileName()));
+      TransformerFactory.newInstance().newTransformer().transform(new DOMSource(document), new StreamResult(output));
+      output.closeEntry();
+    }
+
+    @Override
+    public boolean isEmpty() {
+      return items().count() > 0;
+    }
+
+    protected abstract Stream<T> items();
+
+    protected abstract void populate(Element itemElement, T item);
+
+    protected abstract String xmlCollectionName();
+
+    protected abstract String xmlName();
+
+  }
+
+  private static final DateFormat DF_TIMESTAMP = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+
+  private final String centreName;
+
+  private final Submission submission;
+
+  private final ChildSubmissionFile[] FILES = new ChildSubmissionFile[] { new StudyXmlSubfile(), new SampleXmlSubfile(),
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this field "FILES" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)